.github/workflows/release.yml

```diff
@@ -1,8 +1,14 @@
-name: UX
-on: [push, workflow_dispatch]
+name: Release
```

Comment:
From an outsider, it's a bit confusing to have both a `Release` and `Semantic Release` workflow. Are we able to put this into just a single release workflow and simplify?

Reply:
I'm super open to naming these better. Basically, a commit two master runs one of two workflows:

- If this is a new commit, not a "Version" commit, then run the "Semantic release" workflow. This actually _triggers_ semantic release
- If a tagged commit comes in, then we run "release", which publishes the newly versioned packages to various places.

Otherwise, the two workflows are mostly completely different.

Reply:
Gotcha, thanks for clarifying. Based on this setup wouldn't these two workflows be tightly coupled anyways? Once someone merges a PR or commits a change to the `master` branch it will kick off the Semantic Release workflow, which will in turn kick off the Release workflow once the newly-created tag is pushed up anyways. They're always going to run sequentially in this setup, Semantic Release then Release.

Is there a benefit to splitting this into two workflows instead of just moving the publish steps from the Release workflow to the bottom of the Semantic Release workflow, while adding a `needs` condition to them to force them to run after the semantic release job is finished and publish the new version?

Reply:
> Is there a benefit to splitting this into two workflows instead of just moving the publish steps from the Release workflow to the bottom of the Semantic Release workflow, while adding a needs condition to them to force them to run after the semantic release job is finished and publish the new version?

The reason to have the separate workflow is because they run _after_ the semantic release commit is pushed. So, it would be more awkward to run workflows in the context of a commit that came _after_ the commit that triggered the workflow.

This way, the workflows are easily associated with the correct commits.

Reply:
Makes sense, wasn't sure if there was something I was missing but I can see your reasoning for doing that.

Personally I would change the `Semantic Release` workflow to `Version` and the `Release` workflow to `Publish`. To me, `Release` is an umbrella term, which implies versioning and publishing artifacts. Both of these workflows resemble the whole release process, so I think it makes more sense to just name them after the logical components they represent in the release process.